### PR TITLE
Feature/sort

### LINKED ISSUE #305: ✨Order submissions by Confirmed Plagiarism by default, and provide option to sort similarities
Cc:

Hi 
  
### Pain

In viewing similarities, it is painful traverse over multiple pages to look for all the cases that have been flagged.

### Suggested Solution

Create a sort function that sorts all plagiarism cases (confirmed and suspected) at the top by default, and allows users to sort average similarity, and student names as well.

(Additional) Add a filter function to sieve out students the user would like to view

### Tasks

- [ ] ...
- [ ] ...

### Screenshots

Screenshot from 2023-09-04 13-57-46
Thanks!

### PR DESCRIPTION
**Pull request by mengyewgau**
Added a button on remarks to sort it by descending order and ascending order

## Description

Sorting by descending order (Suspected, Confirmed, Blank) implemented. The priority on sort is always on maximum similarity descending, which indicates how similar both submissions are.

## Related Issue

This PR aims to fix #305, providing the update. The filter function suggested was not implemented, as I felt it might not be very useful in enhancing the user experience, and has no fundamental difference to the sort function

## Motivation and Context

The objective is for reviewers to have a priority ranking on which suspected cases they can review first, based on which is the most similar!

## Screenshots (if appropriate):
Screenshot from 2023-09-11 01-02-46

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,3 +41,6 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
+*.yml
+config/database.yml
```
Comment (huyuxin0429) [inline .gitignore:45]: Why are there two same entries?

**Review by huyuxin0429 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (mengyewgau) [reply .gitignore]: My bad, it might have been automatically added by own configurations. Oversight on my part, will be careful in the future

**Comment by huyuxin0429** [comment]
Due to urgent user request, this functionality will be added to the base repo asap. User wants both max similarity and average similarity to be the top sorted results first

**Comment by knmnyn** [comment]
Thanks for doing this on an urgent basis!

**Comment by huyuxin0429** [comment]
The user requested for the the columns to be sorted with the maximum values of max similarity and average similarity appearing at the top. I interpreted this as performing a multilevel sort, first by avg similarity, then max similarity on the front end UI side. I intend to add those functionality to this PR as it is the most relevant, and merge it into main once completed.

I dont really see much of an reason to sort by submission id, as that does not really give us any useful information.

I have done a raw SQL query directly on the database, here is the query:
```
SELECT U1.name as "Student 1", U2.name as "Student 2", Greatest(SS.similarity_1_to_2, SS.similarity_2_to_1) as "Max Similarity", ((SS.similarity_1_to_2 + SS.similarity_2_to_1)/2) as "Avg Similarity"
FROM ssid_production.submission_similarities as SS
inner join ssid_production.assignments on SS.assignment_id = ssid_production.assignments.id
inner join ssid_production.submissions AS S1 on SS.submission1_id = S1.id
inner join ssid_production.submissions AS S2 on SS.submission2_id = S2.id
inner join ssid_production.courses on ssid_production.assignments.course_id = ssid_production.courses.id
inner join ssid_production.users AS U1 on U1.id = S1.student_id
inner join ssid_production.users AS U2 on U2.id = S2.student_id
WHERE ssid_production.assignments.title = 'Assignment 2'
AND ssid_production.courses.code = 'CS1010E'
order by ((SS.similarity_1_to_2 + SS.similarity_2_to_1)/2) DESC, Greatest(SS.similarity_1_to_2, SS.similarity_2_to_1) DESC
```
image

I'll translate this to the relevant rails query, probably in accordance to https://guides.rubyonrails.org/active_record_querying.html.

**Pull request closed without merging**